### ISSUE TITLE
NoClassDefFoundError: DeleteLinksStatusReport in 5.0.d

### ISSUE THREAD

**Issue by hi-ko** [jgoldhammer/alfresco-jscript-extensions#11]
Which versions of alfresco are supported?
In 5.0.d I get a NoClassDefFoundError:

```
2016-11-02 11:50:28.174  ERROR [web.context.ContextLoader] Context initialization failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'scriptLinkService' defined in URL [jar:file:/opt/alfresco/tomcat-alfresco/webapps/alfresco/WEB-INF/lib/alfresco-jscript-extension-1.3-SNAPSHOT.jar!/alfresco/module/alfresco-jscript-extension/module-context.xml]: Initialization of bean failed; nested exception is java.lang.NoClassDefFoundError: org/alfresco/service/cmr/repository/DeleteLinksStatusReport
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:529)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:458)
        at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:293)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:223)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:290)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:191)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:633)
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:932)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:479)
        at org.springframework.web.context.ContextLoader.configureAndRefreshWebApplicationContext(ContextLoader.java:410)
        at org.springframework.web.context.ContextLoader.initWebApplicationContext(ContextLoader.java:306)
        at org.springframework.web.context.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:112)
        at org.alfresco.web.app.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:63)
        at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:5016)
        at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5528)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:901)
        at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:877)
        at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:652)
        at org.apache.catalina.startup.HostConfig.deployWAR(HostConfig.java:1090)
        at org.apache.catalina.startup.HostConfig$DeployWar.run(HostConfig.java:1900)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: org/alfresco/service/cmr/repository/DeleteLinksStatusReport
        at java.lang.Class.getDeclaredMethods0(Native Method)
        at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
        at java.lang.Class.privateGetPublicMethods(Class.java:2902)
        at java.lang.Class.getMethods(Class.java:1615)
        at org.springframework.beans.ExtendedBeanInfoFactory.supports(ExtendedBeanInfoFactory.java:53)
        at org.springframework.beans.ExtendedBeanInfoFactory.getBeanInfo(ExtendedBeanInfoFactory.java:45)
        at org.springframework.beans.CachedIntrospectionResults.<init>(CachedIntrospectionResults.java:270)
        at org.springframework.beans.CachedIntrospectionResults.forClass(CachedIntrospectionResults.java:186)
        at org.springframework.beans.BeanWrapperImpl.getCachedIntrospectionResults(BeanWrapperImpl.java:321)
        at org.springframework.beans.BeanWrapperImpl.getPropertyDescriptorInternal(BeanWrapperImpl.java:351)
        at org.springframework.beans.BeanWrapperImpl.isWritableProperty(BeanWrapperImpl.java:427)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyPropertyValues(AbstractAutowireCapableBeanFactory.java:1420)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1159)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:519)
        ... 25 more
Caused by: java.lang.ClassNotFoundException: org.alfresco.service.cmr.repository.DeleteLinksStatusReport
        at org.apache.catalina.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1720)
        at org.apache.catalina.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1571)
        ... 39 more
```

**Comment by jgoldhammer** [comment]
Hi Heiko,

I will check that. My idea is to compile with 4.2 and remove all incompatible changes, e.g. removing the DeleteLinksStatusReport. Stay tuned :-)

Jens

**Comment by douglascrp** [comment]
Hey @hi-ko @jgoldhammer 
This seems to be the same thing I reported in the issue #10

**Comment by hi-ko** [comment]
Hi @jgoldhammer , @douglascrp,
sorry - so ticket could be closed as duplicate of #10, but I didn't see a correlation by reading the ticket subject ;-)

**Comment by binduwavell** [comment]
I'm in the process of adding Alfresco JScript Extensions to the generator-alfresco yeoman generator. I'm running into this same issue with SDK 2.1.1, currently testing with community so 5.0.d. I'm able to limit the common amps by SDK version, so as long as it works with SDK 2.2.0 I can make it available there for now. I'm commenting here in hopes that I'll get updates once the issue is resolved :)

**Comment by binduwavell** [comment]
FYI, is also failing with SDK 2.1.1 and enterprise 5.0.1 in the same manner.

**Comment by binduwavell** [comment]
FYI, is working with SDK 2.2.0 community 5.0.e and enterprise 5.1.1.

**Comment by jgoldhammer** [comment]
Thanks for reporting it!
I have created a branch for a version which is compatible with 4.2...
https://github.com/jgoldhammer/alfresco-jscript-extensions/tree/build-4.2
https://github.com/jgoldhammer/alfresco-jscript-extensions/releases/tag/v1.2-4.2

The artifacts are pushed to maven repository. 
Please use version 1.2-4.2 to reference it.

Due to a bug in SDK 3 I can only guarantee that Alfresco 5.0.d is running with my extension, I cannot test it with 4.2 easily... @binduwavell: Can you test it?

@binduwavell Thanks for integrating it into your cool generator project!

**Comment by binduwavell** [comment]
Hi Jens, I can test the amps on alfresco 4.2 and 5.0.1 easily. I will update here if I run into any issues.

**Comment by binduwavell** [comment]
I can confirm that this worked via SDK on 5.0.1. I tried to use MMT to load it into a 4.2.6.6 instance and received this error:

```
Installing AMP '/opt/alfresco-4.2.6.6/amps/jscript.amp' into WAR '/opt/alfresco-4.2.6.6/tomcat/webapps/alfresco.war'
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at com.simontuffs.onejar.Boot.run(Boot.java:287)
	at com.simontuffs.onejar.Boot.main(Boot.java:137)
Caused by: org.alfresco.error.AlfrescoRuntimeException: 10070000 The version string '1.2-4.2' is invalid.
	at org.alfresco.util.VersionNumber.<init>(VersionNumber.java:80)
	at org.alfresco.repo.module.ModuleDetailsImpl.<init>(ModuleDetailsImpl.java:137)
	at org.alfresco.repo.module.tool.ModuleDetailsHelper.createModuleDetailsFromPropertiesStream(ModuleDetailsHelper.java:53)
	at org.alfresco.repo.module.tool.ModuleDetailsHelper.createModuleDetailsFromPropertyLocation(ModuleDetailsHelper.java:83)
	at org.alfresco.repo.module.tool.ModuleManagementTool.installModule(ModuleManagementTool.java:227)
	at org.alfresco.repo.module.tool.ModuleManagementTool.installModules(ModuleManagementTool.java:170)
	at org.alfresco.repo.module.tool.ModuleManagementTool.installModules(ModuleManagementTool.java:153)
	at org.alfresco.repo.module.tool.ModuleManagementTool.main(ModuleManagementTool.java:867)
	... 6 more
```

**Comment by binduwavell** [comment]
I hacked the amp file to update the module.properties to use 1.2.4.2 instead of 1.2-4.2. This allowed the amp to be applied, but then the repo wouldn't start with an error related to 1.2-4.2 module not being loaded.

I had to hack the module.properties in your jar file in the amp.

Once both of these items had been hacked the amp could be applied and the repo started. I was then able to validate with 4.2.6.6 that Alfresco JScript extensions did work. I did not test on a community version of 4.2, but I would expect that to be fine at this point.
